agents/tests/test_agent.py

```diff
@@ -93,6 +73,28 @@ def maybe_restart_service(self):
         return super(MockAgent, self).maybe_restart_service()
 
 
+def setup_interface_test(monkeypatch, net_if_addrs_mock):
+    """Helper function to setup common test environment for interface tests."""
+    def mock_gethostbyname(hostname):
+        raise socket.gaierror('Name or service not known')
+
+    monkeypatch.setattr(psutil, 'net_if_addrs', net_if_addrs_mock)
+    monkeypatch.setattr(socket, 'gethostbyname', mock_gethostbyname)
+
+    agent = MockAgent(port=12345, interface='nonexistent')
+    agent.setup_logging()
+
+    agent.collect_server_metadata()
+
+    assert agent.ip is None
+
+    with open(agent.logfile.name, 'r') as f:
+        f.seek(0)
+        contents = f.read()
+
+    return contents
```

Comment:
Consider renaming to `setup_mock_agent_with_invalid_interface()` or similar for clarity.
Also, extract expected log assertions (e.g., presence of error messages) into the calling test for separation of concerns.